server/src/main/java/org/apache/iotdb/db/engine/merge/selector/MaxFileMergeFileSelector.java

```diff
@@ -228,7 +227,7 @@ private void selectOverlappedSeqFiles(TsFileResource unseqFile) {
         if (seqSelected[i] || !seqFile.getDevices().contains(deviceId)) {
           continue;
         }
-        long seqEndTime = seqFile.getEndTime(deviceId);
+        long seqEndTime = seqFile.isClosed() ? seqFile.getEndTime(deviceId) : Long.MAX_VALUE;
```

Comment:
add a javadoc for TsFileResource.getEndTime()

Reply:
hi ,I have a question.
if a seqfile do not closed. and then  is there a conflict?  flushing and merging...

Reply:
> hi @zhanglingzhe0820 @jixuan1989 @qiaojialin ,I have a question.
> if a seqfile do not closed. and then is there a conflict? flushing and merging...
> 
> 1. merging finished, need delete old tsfile , if this seqfile do not closed,  can this delete ?
> 2. if flushing data  when merging...  , maybe have a chunk data do not write tsfile, but this tsfile merged..
hi
1. unsealed seq file will not be selected to a merge task so it will not be deleted.
2. a file which is flushing is also an unsealed file.

Reply:
that is ok...